charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -16,4 +16,19 @@ data:
   resources.json: |
     {{ toJson .Values.connectInject.apiGateway.managedGatewayClass.resources }}
   {{- end }}
+  {{- if (mustHas "resource-apis" .Values.global.experiments) }}
+  config.yaml: |
+    gatewayClassConfigs:
+      - apiVersion: consul.hashicorp.com/v2beta1
+        metadata:
+          name: consul-mesh-gateway
+          namespace: {{ .Release.Namespace }}
+        deployment:
+          resources:
+            {{ .Values.meshGateway.resources }}
```

Comment:
Is the intention that the entire resource should be present here? If so, this should be `spec.deployment` and have a `kind: GatewayClassConfig`

Reply:
I'm thinking it's likely I just overlooked ^ in the ticket 😆

Reply:
Yes I missed that, but I think I will leave some of the other fields out until they are needed for development. IRT maybe modifying what the GatewayClassConfig looks like

Reply:
Updated